slybot/slybot/plugins/scrapely_annotations/extraction/pageparsing.py

```diff
@@ -25,6 +33,129 @@ def to_template(self, descriptors=None):
                                   self.html_page.page_id, self.ignored_regions,
                                   self.extra_required_attrs, descriptors)
 
+    @staticmethod
```

Comment:
If we are including this much from the `TemplatePageParser` directly, why don't we copy it over entirely? This eases the burden of rebasing it on top of scrapely later on.